metaflow/plugins/kfp/kfp_s3_sensor.py

```diff
@@ -0,0 +1,55 @@
+"""
+This function is called within the s3_sensor_op running container.
+(1) It decodes the path_formatter function's code and runs it to obtain a final S3 path
+(2) It splits the formatted path into an S3 bucket and key 
+(3) It polls for an object with the specified bucket and key until timeout
+"""
+import boto3
+
+
+def wait_for_s3_path(
+    path: str,
+    timeout_seconds: int,
+    polling_interval_seconds: int,
+    path_formatter_code_encoded: str,
+    flow_parameters_json: str,
+) -> None:
+    import boto3
+    import botocore
+    import base64
+    import json
+    import marshal
+    import time
+    from typing import Tuple
+    from urllib.parse import urlparse
+
+    flow_parameters = json.loads(flow_parameters_json)
+    path_formatter_code = marshal.loads(base64.b64decode(path_formatter_code_encoded))
+
+    def path_formatter(key: str, flow_parameters: dict) -> str:
```

Comment:
this is not your path formatter.  This function is ___?

Reply:
Are you suggesting I name the function as `__path_formatter`?

Reply:
Ahh never mind, perhaps a better name would be `path_formatter_template` because it is just the template of the path_formatter, and the code is dynamically added with `path_formatter.__code__=...`.

Reply:
Renamed to `path_formatter_template`.